community/bolt/src/main/java/org/neo4j/bolt/v1/packstream/utf8/SunMiscUTF8Encoder.java

```diff
@@ -0,0 +1,84 @@
+package org.neo4j.bolt.v1.packstream.utf8;
+
+import sun.nio.cs.ArrayEncoder;
+
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.reflect.Field;
+import java.nio.ByteBuffer;
+import java.nio.charset.CharsetEncoder;
+import java.nio.charset.StandardCharsets;
+
+public class SunMiscUTF8Encoder implements UTF8Encoder
+{
+    private final CharsetEncoder charsetEncoder = StandardCharsets.UTF_8.newEncoder();
+    private final ArrayEncoder arrayEnc = (ArrayEncoder) charsetEncoder;
+
+    private final byte[] out = new byte[1024 * 16];
```

Comment:
Since there'll be one of these per session, I didn't want to take up too much. But I'll add a toggle to set it higher to the PR